api/lib/api.py

```diff
@@ -0,0 +1,252 @@
+    game = response.get_game()
+
+    moves = query.get_moves_string(game)
+
+    # resign the player and claim victory
+    game_utils.complete_game(game)
+
+    if is_pvp:
+        return game_utils.pvp_claim_victory(moves, mover_p, opponent_p, stockfish_app)
+    else:
+        return game_utils.ai_claim_victory(moves, mover_p, stockfish_app)
+
+# allow the player to cheat
+@root.route('/cheat',methods = ['POST', 'GET'])
+def cheat():
+    args = utils.parse_args(request)
+    mover = utils.get_first_valid_index(args, ['self'])
+    opponent = utils.get_first_valid_index(args, ['opponent'])
+    is_ai = utils.get_first_valid_index(args, ['ai'])
+
+    response = game_utils.get_game(mover, opponent, is_ai)
+    if not response.is_valid():
+        return response.get_error()
+    
+    game = response.get_game()
+    moves = query.get_moves_string(game)
+
+    # return cheats
+    board, eval, best = chessboard.cheat_board(logger, stockfish_app, moves)
+
+    return utils.respond(f'{board}\n{eval}\n{best}\nthat good enough for you? stupid cheater {utils.mention_player(mover)}', 200)
+
+# Startup code
+CONFIG_PATH_DEST = "configPath"
+
+def command_line_parse(argv = sys.argv[1:]):
+    description = """Chess Move Restful API"""
+    
+    argparser = argparse.ArgumentParser(description=description)
+    argparser.add_argument("--file", "--config", dest=CONFIG_PATH_DEST, default="./config.json", help = "File Path to Config File")
+    argparser.add_argument("--db", "--database", dest="db", help="SQL URI for accessing SQL database")
+    return vars(argparser.parse_args(argv))
+
+def create_app(args={}):
+    config = {}
+
+    # There is also config file loading that comes with json from flask's library
+    #   especially if you don't like this hacky code
+    data = {}
+    if CONFIG_PATH_DEST in args:
+        print(f"Loading config file from {args[CONFIG_PATH_DEST]}")
+        with open(args[CONFIG_PATH_DEST], 'r') as configFile:
+            data = json.load(configFile)
+
+    # From Config File
+    utils.safe_dict_copy_default(config, ['host'], data, ['api', 'host'], "0.0.0.0")
+    utils.safe_dict_copy_default(config, ['port'], data, ['api', 'port'], 8000)
+    utils.safe_dict_copy_default(config, ['stockfish-path'], data, ['api', 'stockfish', 'path'], '/usr/games/stockfish')
+    utils.safe_dict_copy_default(config, ['stockfish-depth'], data, ['api', 'stockfish', 'depth'], 15)
+    utils.safe_dict_copy_default(config, ['stockfish-params'], data, ['api', 'stockfish'], {})
+    utils.safe_dict_copy_default(config, ['flask'], data, ['api', 'flask'], {})
+    utils.safe_dict_copy(config, ['db'], data, ['api', 'db_uri'])
+
+    # From CLI Args
+    utils.safe_dict_copy_default(config, ['db'], args, ['db'], "sqlite:///storage.db")
+
+    query.init(config['db'])
+    global stockfish_app
+    stockfish_app = Stockfish(path=config['stockfish-path'], depth=config['stockfish-depth'], parameters=config['stockfish-params'])
```

Comment:
> why is it preferred to pass in a stockfish instance to all api.py methods as opposed to storing a single stockfish instance in utils.py as before? stockfish isn't actually used (and shouldn't be) in api.py anyways

In order to configure how the stockfish app is used we need to be able to control its lifetime. This would mean either making the chessboard (utils) module an instance and creating that or adding it to a module that already has an instance.

I prefer loading all code prior to performing runtime/process. In the previous version of the code we were starting the stockfish instance on import. We could instead make a chessboard object and just call all helper functions off that instance instead of making static references to helper functions. Would that be better?